tests/test_mab.py

```diff
@@ -1767,3 +1767,131 @@ def test_pickle_fitted_reproducibile_expectations(self):
                 self.assertListEqual(pe1, pe2)
 
         os.remove('mab.pkl')
+
+    #################################################
+    # Test arm_status
```

Comment:
Thanks for these tests. Maintaining arm_status correctly is non-trivial in mind. Since a "mab" object is a living object, many things can happen on top if.. later in the process, having arm_status to still be correct requires some good testing at this stage possibly with interesting usage scenarios.

Reply:
Just to set expectations; I expect to run into bugs in future where we could not track arm_status correctly.

Reply:
Similar concerns. I'll add some more tests.

Reply:
and I am totally OK with that expectation.. The only issue/worry, if used in PROD, then such bugs will come back to us.. Some calculated risk, I guess..